src/picongpu/include/plugins/ResourceLog.hpp

```diff
@@ -0,0 +1,243 @@
+        ResourceMonitor<simDim> resourceMonitor;
+
+        // programm options
+        std::string outputFilePrefix;
+        std::string streamType;
+        std::string outputFormat;
+        std::vector<std::string> properties;
+
+        std::filebuf fileBuf;
+        std::map<std::string, bool> propertyMap;
+
+    public:
+
+        ResourceLog() :
+                cellDescription(NULL)
+        {
+            Environment<>::get().PluginConnector().registerPlugin(this);
+        }
+
+        std::string pluginGetName() const
+        {
+            return "ResourceLog";
+        }
+
+        void notify(uint32_t currentStep)
+        {
+            //
+            // Create property tree which contains the resource information
+            using boost::property_tree::ptree;
+            ptree pt;
+
+            if(contains(propertyMap, "rank"))
+            {
+                size_t rank = static_cast<size_t>(Environment<simDim>::get().GridController().getGlobalRank());
+                pt.put("resourceLog.rank", rank);
+            }
+
+            if(contains(propertyMap,"position"))
+            {
+                DataSpace<simDim> currentPosition = Environment<simDim>::get().GridController().getPosition();
+                pt.put("resourceLog.position.x", currentPosition[0]);
+                pt.put("resourceLog.position.y", currentPosition[1]);
+                pt.put("resourceLog.position.z", currentPosition[2]);
+            }
+
+            if(contains(propertyMap, "currentStep"))
+            {
+                pt.put("resourceLog.currentStep", currentStep);
+            }
+
+            if(contains(propertyMap, "cellCount"))
+            {
+                size_t cellCount = resourceMonitor.getCellCount();
+                pt.put("resourceLog.cellCount", cellCount);
+            }
+
+            if(contains(propertyMap,"particleCount"))
+            {
+                std::vector<size_t> particleCounts = resourceMonitor.getParticleCounts<VectorAllSpecies>(*cellDescription);
+                pt.put("resourceLog.particleCount", std::accumulate(particleCounts.begin(), particleCounts.end(), 0));
+            }
+
+            //
+            // Write property tree to string stream
+            std::stringstream ss;
+            if(outputFormat == "json")
+            {
+                write_json(ss, pt, false);
+            }
+            else if(outputFormat == "jsonpp")
+            {
+                write_json(ss, pt, true);
+            }
+            else if(outputFormat == "xml")
+            {
+                write_xml(ss, pt);
+            }
+            else if(outputFormat == "xmlpp")
+            {
+                write_xml(ss, pt, boost::property_tree::xml_writer_make_settings<std::string>('\t', 1));
+            }
+            else
+            {
+                throw std::runtime_error(std::string("resourcelog.format ") + outputFormat + std::string(" is not known, use json or xml."));
```

Comment:
please add `xmlpp` and `jasonpp` to the comment since they can be used as well. 
e.g. `" is not known, use json(pp) or xml(pp)."`